### ISSUE TITLE
Disable "Automatically redirect .onion sites" when users disable "Private window with Tor"

### ISSUE THREAD

**Issue by Tonev** [brave/brave-browser#17363]
## Description 
If enabled, `Automatically redirect .onion sites` will remain in an enabled state and grayed out upon disabling `Private window with Tor` from any existing user profile. 

I agree it's not a big deal to disable `Automatically redirect .onion sites` before `Private window with Tor`, but when using multiple user profiles, only `Private window with Tor` is shared between the user profiles, so it's quite possible `Automatically redirect .onion sites` to remain in an enabled state and grayed out in some user profile, and users will have to re-enable `Private window with Tor` in order to disable `Automatically redirect .onion sites`, and then disable both in the correct order.

## Steps to Reproduce

   1. Go to `brave://settings/extensions`.
   2. Enable `Private window with Tor` and `Automatically redirect .onion sites`.
   3. Disable `Private window with Tor`.

## Actual result:

![actual](https://user-images.githubusercontent.com/20608253/128377389-9c0febeb-74a7-4251-bc47-a6e9e521bb7b.png)

## Expected result:
![expected](https://user-images.githubusercontent.com/20608253/128377459-53723efa-4f4a-40fa-bb57-641dbec753cc.png)

## Reproduces how often: 

Easily

## Brave version (brave://version info)

<html>
<body>

Brave | 1.27.109 Chromium: 92.0.4515.115 (Official Build) (64-bit)
-- | --
Revision | 48cb2f4029b84b003719740a6cf9ca73f374a857-refs/branch-heads/4515_105@{#4}
OS | Windows 7 Service Pack 1 (Build 7601)
JavaScript | V8 9.2.230.20
User Agent | Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.115 Safari/537.36
Command Line | "C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe" --enable-dom-distiller --disable-domain-reliability --no-pings --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --enable-features=LegacyTLSEnforced,WebUIDarkMode,AutoupgradeMixedContent,WinrtGeolocationImplementation,SafetyTip,PrefetchPrivacyChanges,PasswordImport,ReducedReferrerGranularity --disable-features=FirstPartySets,AutofillServerCommunication,InterestCohortFeaturePolicy,FledgeInterestGroups,HandwritingRecognitionWebPlatformApi,FlocIdComputedEventLogging,HandwritingRecognitionWebPlatformApiFinch,PrivacySandboxSettings,FederatedLearningOfCohorts,TrustTokens,DirectSockets,IdleDetection,WebOTP,NotificationTriggers,FledgeInterestGroupAPI,AutofillEnableAccountWalletStorage,InterestCohortAPIOriginTrial,NetworkTimeServiceQuerying,SubresourceWebBundles,SignedExchangePrefetchCacheForNavigations,LangClientHintHeader,SignedExchangeSubresourcePrefetch,LiveCaption,EnableProfilePickerOnStartup,TextFragmentAnchor,EnablePasswordsAccountStorage --flag-switches-begin --enable-features=LegacyTLSEnforced,WebUIDarkMode,AutoupgradeMixedContent,WinrtGeolocationImplementation,SafetyTip,PrefetchPrivacyChanges,PasswordImport,ReducedReferrerGranularity,EphemeralStorage --flag-switches-end
Executable Path | C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe
Profile Path | C:\Users\User\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default
Variations | AdRewardsStudy:NextPaymentDayEphemeralStorageStudy:EnabledNativeCosmeticFilteringStudy:EnabledPermissionLifetimeReleaseStudy:Enabled

</body>
</html>

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by rebron** [comment]
@Tonev I don't see this as an issue since you need to enable `Private Window with Tor` for `Automatically redirect .onion sites` to work. If `Private Window with Tor` is disabled anywhere, `Automatically redirect to .onion sites` just won't work. In the UX it's greyed out properly and it's not enabled.

I understand the suggestion of tying the Private Window with Tor toggles together on the disable but the UX to grey out the `Automatically redirect to .onion sites` does that versus toggling off. Turning one toggle off and then you see two toggles move makes things less predictable. 

Let me know if I'm misunderstanding. I think this is a wontfix.

As you mentioned, https://github.com/brave/brave-browser/issues/15199#issuecomment-820687902 we do want to address this issue though.

**Comment by Tonev** [comment]
@rebron 

It's not really an issue. Just thought tying both options when users disable `Private window with Tor` is a better alternative than potentially leaving `Automatically redirect .onion sites` in an "enabled" but grayed out state (I'm aware the feature isn't active).

My concrete example would be the following:

- 2 user profiles -> `Profile 1` and `Profile 2`.
- `Profile 1` enables both `Private window with Tor` and `Automatically redirect .onion sites`, which automatically enables `Private window with Tor` for `Profile 2`. 
- If `Profile 2` decides to disable `Private window with Tor` at some point, the option will be disabled in both `Profile 1` and `Profile 2`, but `Automatically redirect .onion sites` will remain in an "enabled" but grayed out state in `Profile 1`.

If you decide to implement my suggestion in the future, `Profile 1` wouldn't have to re-enable `Private window with Tor` in order to set `Automatically redirect .onion sites` to a completely disabled state, and then disable `Private window with Tor` once again.

Long story short, although it's not an issue (the feature isn't active), it would make more sense to automatically disable `Automatically redirect .onion sites` the moment `Private window with Tor` is disabled since the first option can't work without the latter, leaving it in an "enabled" but grayed out state is unnecessary, at least in my opinion.

**Comment by rebron** [comment]
@Tonev I'm going to close this one out as wontfix. This is a ux choice where the settings are linked but the toggle to disable the first one grays out/disables the second one versus toggling off the first setting toggles off both. 

It's a good catch though. 

cc: @karenkliu can you reopen if you think differently.
